### DISCUSSION TITLE
Feature request: more quick-terminal options are required

### DISCUSSION THREAD

**Discussion by pabroux** [ghostty-org/ghostty#3717] [Feature Requests, Ideas] [answered]
Hello,

When I create a quick top terminal, the window takes the full width of the screen. If you have a big monitor (e.g. 32' or 49'), it's terrible:

![Untitled](https://github.com/user-attachments/assets/ba5a4955-6c8d-4b21-8df1-94b4410bb9d1)

Even if you have a window manager which may add a left/right padding (e.g. [yabai](https://github.com/koekeishiya/yabai)), that quick terminal window is not taken into account compared to other Ghostty windows (as you can see in the screenshot above).

1. Is it possible to add an option to display the quick terminal inside a Ghoosty window instead of the screen?
2. Is it possible to set a max length/width of the quick terminal window?

_p.s.: Thank you for your great work by the way. Ghostty is great!_

**Comment by JakeWharton** [comment, marked as answer]
Sizing options, including in two dimensions (i.e., not just height config) are mentioned in https://github.com/ghostty-org/ghostty/issues/2384

**Comment by pabroux** [comment]
It seems someone proposed a PR #3742 with the new option `quick-terminal-size`.
